p2pv2/stream_pool.go

```diff
@@ -152,7 +153,7 @@ func (sp *StreamPool) sendMessage(ctx context.Context, msg *p2pPb.XuperMessage,
 		return err
 	}
 	if err := str.SendMessage(ctx, msg); err != nil {
-		if err.Error() == "stream reset" {
+		if err.Error() == "stream reset" || strings.HasSuffix(err.Error(), "connection reset by peer") {
```

Comment:
Using error string to check for errors is not an idiomatic way.

Reply:
Yes!